src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -289,13 +333,18 @@ public void postInject() {
 
         logger.dfltKv("ThingName", (Supplier<String>) () ->
                 Coerce.toString(deviceConfiguration.getThingName()));
-
-        executorService.submit(() -> {
-            subscribeToJobsTopics();
-            logger.atInfo().log("Connection established to IoT cloud");
-            deploymentStatusKeeper.publishPersistedStatusUpdates(DeploymentType.IOT_JOBS);
-            this.fleetStatusService.updateFleetStatusUpdateForAllComponents();
-        });
+        if (subscriptionFuture != null && !subscriptionFuture.isDone()) {
+            subscriptionFuture.cancel(false);
```

Comment:
always use `true`. You must interrupt the existing thread, otherwise it won't stop